modules/standard/Buffers.chpl

```diff
@@ -306,16 +308,65 @@ module Buffers {
      :arg error: (optional) capture an error that was encountered instead of
                   halting on error
    */
-  proc buffer.buffer(out error:syserr) {
+  proc buffer.init(out error:syserr) {
     this.home = here;
+    super.init();
     error = qbuffer_create(this._buf_internal);
   }
   pragma "no doc"
-  proc buffer.buffer() throws {
+  proc buffer.init() /*throws*/ {
     var error:syserr = ENOERR;
     this.home = here;
+    super.init();
     error = qbuffer_create(this._buf_internal);
-    if error then try ioerror(error, "in buffer constructor");
+    if error then try! ioerror(error, "in buffer constructor");
```

Comment:
I'd leave a note on this function about wanting to eventually allow it to throw when you can do that from initializers.

Reply:
I was going to open an issue against that and rely on the commented out `throws` as a placeholder for the intention, but I'll add a comment to this line as well.

Reply:
#6145 has the most recently documented discussion of that, but I believe the thinking has advanced beyond that.  You'll want to ask @psahabu